Makefile

```diff
@@ -1,17 +1,24 @@
 .POSIX:
-.PHONY: all compile test clean check
+.PHONY: all compile test clean check install
 .SUFFIXES: .el .elc
 
+DESTDIR ?=
+PREFIR  ?= /usr/local
+DATADIR ?= $(PREFIX)/data
+BINDIR  ?= $(PREFIX)/bin
+LIBDIR  ?= $(PREFIX)/lib
+INCDIR  ?= $(PREFIX)/include/$(NAME)
+ELDIR   ?= $(DATADIR)/emacs/site-lisp
+
+
 EMACS = emacs
 MAKEINFO = makeinfo
 BYTEC = compat-25.elc \
 	compat-26.elc \
 	compat-27.elc \
 	compat-28.elc \
 	compat-29.elc \
-	compat.elc \
-	compat-macs.elc \
-	compat-tests.elc
+	compat.elc
```

Comment:
This change is no valid. We have to byte compile all the files for CI checking.

Reply:
Even the macros? I thought the macros are only used inside the other files during byte compilation? (as the comment inside it states).

Reply:
Why do you byte-compile compat-macs separately when it is only used during compilation inside the other files? It doesn't make sense.

Reply:
Yes, they are. It is still useful to have the files available at runtime for development, since this allows you to expand the macro definitions. Compiling in the CI is needed such that we can take advantage of the byte compiler warnings. Byte compiler warnings help catching bugs early on.

Reply:
I added the tests back to be byte-compiled.

Reply:
The invalid commit https://github.com/emacs-compat/compat/pull/30/commits/383cf9683deb500d9d18e52b26f7f4758710b9cf is still there. Did you accidentially force-push it again?

Reply:
> To be clear, BYTEC must not be changed. The variable should stay as is.

If the other files are dependencies of `compile:` they will be byte-compiled.
Just run the ci or run make compile locally and see they will be compiled.

Reply:
> No, your change is still not good, since the `make clean` target is also relying on BYTEC. See @phikal's comment below on how to resolve this in a better way.

BYTEC works the same way as before. Just try the patch and run make compile and make clean there's no difference.

Reply:
> BYTEC works the same way as before. Just try the patch and run make compile and make clean there's no difference.

Yes, now it does. When I commented on this before, it did not.